File: samples/java/Bingo/Server/Play/src/main/java/systems/zlink/samples/bingo/server/play/infrastructure/zlink/spots/entryspot/handlers/ObserveBingoEventsHandler.java
package systems.zlink.samples.bingo.server.play.infrastructure.zlink.spots.entryspot.handlers;

import systems.zlink.framework.ZLinkMessageContext;
import systems.zlink.framework.spots.ZLinkEntrySpotActorRequestHandler;
import systems.zlink.samples.bingo.server.play.infrastructure.zlink.actors.PlayerActor;
import systems.zlink.samples.bingo.server.play.infrastructure.zlink.spots.entryspot.BingoEntrySpot;
import systems.zlink.samples.bingo.shared.contracts.Messages;

import java.util.concurrent.CompletionStage;

public final class ObserveBingoEventsHandler
        implements ZLinkEntrySpotActorRequestHandler<
                BingoEntrySpot,
                PlayerActor,
                Messages.ObserveBingoEventsReq,
                Messages.ObserveBingoEventsRes> {
    @Override
    public CompletionStage<Messages.ObserveBingoEventsRes> handle(
            BingoEntrySpot spot,
            PlayerActor actor,
            ZLinkMessageContext context,
            Messages.ObserveBingoEventsReq request) {
        return spot.observeEvents(actor, request);
    }
}

File: samples/kotlin/Bingo/Server/Play/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/play/infrastructure/zlink/spots/entryspot/handlers/ObserveBingoEventsHandler.kt
package systems.zlink.samples.kotlin.bingo.server.play.infrastructure.zlink.spots.entryspot.handlers

import systems.zlink.framework.ZLinkMessageContext
import systems.zlink.framework.kotlin.ZLinkSuspendingEntrySpotActorRequestHandler
import systems.zlink.samples.kotlin.bingo.server.play.infrastructure.zlink.actors.PlayerActor
import systems.zlink.samples.kotlin.bingo.server.play.infrastructure.zlink.spots.entryspot.BingoEntrySpot
import systems.zlink.samples.kotlin.bingo.shared.contracts.ObserveBingoEventsReq
import systems.zlink.samples.kotlin.bingo.shared.contracts.ObserveBingoEventsRes

class ObserveBingoEventsHandler :
    ZLinkSuspendingEntrySpotActorRequestHandler<
        BingoEntrySpot,
        PlayerActor,
        ObserveBingoEventsReq,
        ObserveBingoEventsRes,
    > {
    override suspend fun handle(
        entrySpot: BingoEntrySpot,
        actor: PlayerActor,
        context: ZLinkMessageContext,
        request: ObserveBingoEventsReq,
    ): ObserveBingoEventsRes = entrySpot.observeEvents(actor, request)
}
